### PR TITLE
adding ecs-service modules

### PR DESCRIPTION
**Pull request by minniux**
adding ecs-service modules, one without LB, and one with elb of only ssl, no s3 logs type

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,3 +53,52 @@ module "service-scaling" {
...
+```
+module "new_service" {
+  source              = "https://github.com/skyscrapers/terraform-ecs//ecs-service"
+  service_name        = "${var.service_name}"
+  template_data       = "${data.template_file.task.rendered}"
+  ecs_cluster         = "${data.terraform_remote_state.ecs.ecs_cluster}"
+}
+```
+
+## ecs-service
```
Comment (ringods) [inline README.md:73]: Shouldn't this be `ecs-service-with-elb`?

File: ecs-service_with_elb/main.tf [hunk 2]
```diff
@@ -0,0 +1,45 @@
+resource "aws_ecs_task_definition" "definition" {
+  family                = "${var.service_name}_${terraform.workspace}"
+  container_definitions = "${var.template_data}"
+  network_mode          = "bridge"
+  task_role_arn         = "${aws_iam_role.role.arn}"
+}
+
+
+module "elb" {
```
Comment (SamClinckspoor) [inline ecs-service_with_elb/main.tf:9]: Why is this needed? afaik if you configure ECS with an ALB you can register tasks in it.
The service doesn't need to be tied to the ELB/ALB creation.

File: ecs-service_with_elb/main.tf [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  source                    = "github.com/skyscrapers/terraform-loadbalancers//elb_only_ssl_no_s3logs"
```
Comment (ringods) [inline ecs-service_with_elb/main.tf:10]: Please lock the module to a specific tag!

**Review by ringods — CHANGES_REQUESTED** [review]
Please don't mix hyphens en underscores in the name. For the moment, we mostly used hyphens.

`ecs-service_with_elb` -> `ecs-service-with-elb`

**Comment by venturel** [comment]
I cannot think about any use-case where I'd use those modules specially the one with an ELB

File: ecs-service_with_elb/main.tf (hunk 2 above)
Comment (MattiasGees) [reply ecs-service_with_elb/main.tf:9]: I would indeed rather use ALB than ELB with ECS.

File: ecs-service/main.tf [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  container_definitions = "${var.template_data}"
+  network_mode          = "bridge"
+  task_role_arn         = "${aws_iam_role.role.arn}"
+}
+
+resource "aws_ecs_service" "container" {
+  name            = "${var.service_name}_${terraform.workspace}"
+  cluster         = "${var.ecs_cluster}"
+  task_definition = "${aws_ecs_task_definition.definition.arn}"
+  desired_count   = 1
```
Comment (MattiasGees) [inline ecs-service/main.tf:12]: Should this be hardcoded?

File: ecs-service/variables.tf [hunk 5]
```diff
@@ -0,0 +1,9 @@
+variable "aws_region" {
```
Comment (MattiasGees) [inline ecs-service/variables.tf:1]: Can you add a description for all variables?

**Comment by MattiasGees** [comment]
I commented on this PR, but I have to follow @venturel that I am not convinced this should be made general. IMO this is most of the times quit application specific config. But I am happy to be proven otherwise ;)

File: ecs-service_with_elb/main.tf (hunk 2 above)
Comment (minniux) [reply ecs-service_with_elb/main.tf:9]: @SamClinckspoor You mean ECS creates the ELB/ALB already automatically when load balancer is specified? 
I didn't work with ECS before, so maybe there's something I'm missing, but the load balancer info in here made me think the LB needs to be created separately: 
https://www.terraform.io/docs/providers/aws/r/ecs_service.html#load_balancer
Specially qouting this from http://docs.aws.amazon.com/AmazonECS/latest/developerguide/create-standard-load-balancer.html
```
After your load balancer is created, you can specify it in a service definition when you create a service. For more information, see Creating a Service.
```

Comment (MattiasGees) [reply ecs-service_with_elb/main.tf:9]: @minniux your loadbalancer is not created automatically but we have to do it. But I prefer using an ALB which you can use for multiple services/applications. So the loadbalancer is not really tied to one service.

Also by using ALB it makes the registration to ECS really easy. Happy to explain this more in depth how this works if you want.

**Comment by minniux** [comment]
@venturel @MattiasGees I've to invite @ringods to the discussion  as he suggested moving it to here 

Personally I don't think it's application-specific, I made it as generic as possible to cover just an ecs service for tasks that runs a web server and another for ones that doesn't. But I'd of course leave it to the runtime circle members to say if that's really needed/used for cases except the one on hand

File: ecs-service_with_elb/main.tf (hunk 2 above)
Comment (MattiasGees) [reply ecs-service_with_elb/main.tf:9]: @minniux all the different options. http://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-load-balancing.html

As you can see ALB is much more flexible.

**Comment by MattiasGees** [comment]
I don't think the ELB belongs in this module. As it is not efficient and like I made clear in these comments:
- https://github.com/skyscrapers/terraform-ecs/pull/9#discussion_r154052642
- https://github.com/skyscrapers/terraform-ecs/pull/9#discussion_r154062058

I believe more in ALB and coverage for more applications (same as ingress controller in K8s).

I also think the almost everything needs to be put as a variable (even the blocks) and don't feel it than makes sense to create a module for it.

**Comment by ringods** [comment]
@minniux as I mentioned in [this Slack message](https://skyscrapers.slack.com/archives/C04P0T47V/p1512045726000296), I don't agree with a module calling another module. Leave that up to the stack to combine modules. So my suggestion is to drop the `ecs-service-with-elb` completely.

**Comment by ringods** [comment]
@minniux you might want to close this PR and clean up the branch. ;-)
